From 855cfa63445109f3dd5cd8b1ef0b4e26b6081855 Mon Sep 17 00:00:00 2001
From: Sarika Sharma <quic_sarishar@quicinc.com>
Date: Tue, 19 Dec 2023 11:04:44 +0530
Subject: [PATCH] wifi: ath12k: add pubsta default value to null

Add default value of pubsta as NULL in ath12k_dp_rx_deliver_msdu().
Since pubsta can be used if peer does not exist.
Hence pubsta will not be fetched always.

Signed-off-by: Sarika Sharma <quic_sarishar@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/dp_rx.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/wireless/ath/ath12k/dp_rx.c b/drivers/net/wireless/ath/ath12k/dp_rx.c
index 69b4e46..4fb2537 100644
--- a/drivers/net/wireless/ath/ath12k/dp_rx.c
+++ b/drivers/net/wireless/ath/ath12k/dp_rx.c
@@ -3234,7 +3234,7 @@ static void ath12k_dp_rx_deliver_msdu(struct ath12k *ar, struct napi_struct *nap
 	};
 	struct ieee80211_radiotap_he *he;
 	struct ieee80211_rx_status *rx_status;
-	struct ieee80211_sta *pubsta;
+	struct ieee80211_sta *pubsta = NULL;
 	struct ath12k_peer *peer = NULL;
 	struct ath12k_skb_rxcb *rxcb = ATH12K_SKB_RXCB(msdu);
 	u8 decap = DP_RX_DECAP_TYPE_RAW;
-- 
2.34.1

